opslevel/datasource_opslevel_property_definition.go

```diff
@@ -0,0 +1,50 @@
+package opslevel
+
+import (
+	"github.com/opslevel/opslevel-go/v2023"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourcePropertyDefinition() *schema.Resource {
+	return &schema.Resource{
+		Read: wrap(datasourcePropertyDefinitionRead),
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:        schema.TypeString,
+				Description: "The id of the property definition to find.",
+				ForceNew:    true,
+				Optional:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The display name of the property definition.",
+				Computed:    true,
+			},
+			"schema": {
+				Type:        schema.TypeString,
+				Description: "The schema of the property definition.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func datasourcePropertyDefinitionRead(d *schema.ResourceData, client *opslevel.Client) error {
+	id := d.Get("id").(string)
+	resource, err := client.GetPropertyDefinition(id)
```

Comment:
Don't property definitions have both an ID and Alias.  If so we should support lookup by both.

This means you should modify the schema to make `id` a computed field, add `aliases` and then add a `identifier` as the only input a customer supplies.